addon/record-data.js

```diff
@@ -549,9 +549,11 @@ export default class M3RecordData {
   }
 
   removeFromRecordArrays() {
-    this._recordArrays.forEach(recordArray => {
-      recordArray._removeRecordData(this);
-    });
+    if (CUSTOM_MODEL_CLASS) {
+      this._recordArrays.forEach(recordArray => {
+        recordArray._removeRecordData(this);
+      });
+    }
```

Comment:
Why do we only call this when `CUSTOM_MODEL_CLASS` is on?

If the answer is b/c `removeFromRecordArrays` is only called under such circumstances we should assert that

Reply:
:+1: will add the assertion